src/website/src/app/documentation/demos/datagrid/single-selection/single-selection.ts

```diff
@@ -50,13 +58,41 @@ export class DatagridSelectionSingleDemo {
   example = EXAMPLE;
   rowSelectionExample = ROW_SELECTION_EXAMPLE;
   selectionChangeEventExample = SELECTION_CHANGE_EVENT_EXAMPLE;
+  unselectableRow = UNSELECTABLE_ROW;
   users: User[];
   singleSelected: User;
   rowSelected: User;
+  lockedUsers: User[] = [];
 
   constructor(inventory: Inventory) {
     inventory.size = 10;
     inventory.reset();
     this.users = inventory.all;
+
+    this.lockedUsers = [...inventory.all];
+  }
+
+  unlockRow() {
+    this.lockedUsers = this.lockedUsers.map(row => {
+      row.locked = undefined;
+      return row;
+    });
+  }
+
+  lockRow() {
+    // reset selections
+    this.rowSelected = undefined;
+
+    this.lockedUsers = this.lockedUsers.map((user, index) => {
+      /** lock some rows */
+      if ([2, 3, 5, 10].includes(index)) {
+        user.locked = true;
+      }
+      /** also mark them as selected */
+      if ([5].includes(index)) {
```

Comment:
Remove this, it is confusing to me trying to use the demo for the selection to change on me.